frame/democracy/src/lib.rs

```diff
@@ -227,39 +229,48 @@ pub trait Trait: system::Trait + Sized {
 
 	/// Period in blocks where an external proposal may not be re-submitted after being vetoed.
 	type CooloffPeriod: Get<Self::BlockNumber>;
+
+	/// The amount of balance that must be deposited per byte of preimage stored.
+	type PreimageByteDeposit: Get<BalanceOf<Self>>;
+
+	/// Handler for the unbalanced reduction when slashing a preimage deposit.
+	type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
 }
 
 /// Info regarding an ongoing referendum.
 #[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
-pub struct ReferendumInfo<BlockNumber: Parameter, Proposal: Parameter> {
+pub struct ReferendumInfo<BlockNumber: Parameter, Hash: Parameter> {
 	/// When voting on this referendum will end.
 	end: BlockNumber,
-	/// The proposal being voted on.
-	proposal: Proposal,
+		/// The hash of the proposal being voted on.
```

Comment:
```suggestion
	/// The hash of the proposal being voted on.
```

GitHub white spacing rendering is an issue.... sorry :/